slashid/javascript
# [SID-1497] Password support

## Description

[YouTrack issue](https://todo.irdcorp.dev/issue/SID-1497)

This PR removes the `jump-page` app as it was moved to `ng-evangelion`. The main reason was to simplify deployments and version management, since the `jump-page` depends directly on the code SDK and needs to be released at the same time.

To do this, the `react-primitives` package was once again exposed through `npm`, but in a specific way:
- source files are being kept in the package
- only the type declarations are built

This way the consumer of the package can build the app autonomously, which is important because of the way we use `vanilla-extract`.

The biggest TODO left is now writing automated tests.

## Testing guide

- first make sure you check out the branch `sid-1437-password-reset` in the `ng-evangelion` repo
- run `task up` there
- verify that you are targeting the local API in `javascript/packages/react`
- in `javascript` run `pnpm dev --filter react`
- test the password flow in the top left form

## Checklist

- [ ] I have generated the new version of the docs website and smoke tested it
- [ ] I have checked that my changes haven't caused semantic errors in the existing docs
- [ ] I have generated a `changeset` if my change affects the published packages
